### PR TITLE
Add authentication to AppInterface request

### PR DESCRIPTION
**Pull request by mprpic**
See also ops repo MR 174.

Resolves: CORGI-734

### CODE REVIEW

**Comment by mprpic** [comment]
@RedHatProductSecurity/corgi-devs please review.

File: corgi/collectors/app_interface.py [hunk 1]
```diff
@@ -42,6 +42,7 @@ def fetch_service_metadata(
         """
         response = requests.post(
             f"{settings.APP_INTERFACE_URL}/graphql",
+            auth=(settings.APP_INTERFACE_USERNAME, settings.APP_INTERFACE_PASSWORD),
```
Comment (juspence) [inline corgi/collectors/app_interface.py:45]: Does app interface support any kind of cert-based auth or auth with Kerberos tickets? Lots of other RH services seem to be deprecating simple username / password-based auth interfaces to meet the ESS requirements.

It's a pain to have to switch these later or watch stuff break, so if possible I'd rather just jump straight to an auth option that's (hopefully) going to hang around for a while.

**Review by juspence — APPROVED** [review]
LGTM!

File: corgi/collectors/app_interface.py (hunk 1 above)
Comment (mprpic) [reply corgi/collectors/app_interface.py:45]: No other option exists.

**Pull request merged**